test/integration.test.ts

```diff
@@ -0,0 +1,219 @@
+import Ganache from "ganache";
+import { ZeroAddress, ethers, getBytes, isHexString, toUtf8Bytes, AbiCoder } from "ethers";
+import { describe, it, expect, beforeAll, afterAll } from "@jest/globals";
+import { keccak_256 } from "@noble/hashes/sha3";
+import { MockBlocklockReceiver__factory, BlocklockSender__factory, BlocklockSignatureScheme__factory, DecryptionSender__factory, SignatureSchemeAddressProvider__factory } from "../src/generated"
+import { SolidityEncoder, Blocklock, encodeCiphertextToSolidity, parseSolidityCiphertextString, extractSingleLog } from "../src"
+import { BlsBn254 } from "../src/crypto/bls-bn254";
+import { IbeOpts, preprocess_decryption_key_g1 } from "../src/crypto/ibe-bn254";
+
+const blocklock_default_pk = {
+  x: {
+    c0: BigInt("0x2691d39ecc380bfa873911a0b848c77556ee948fb8ab649137d3d3e78153f6ca"),
+    c1: BigInt("0x2863e20a5125b098108a5061b31f405e16a069e9ebff60022f57f4c4fd0237bf"),
+  },
+  y: {
+    c0: BigInt("0x193513dbe180d700b189c529754f650b7b7882122c8a1e242a938d23ea9f765c"),
+    c1: BigInt("0x11c939ea560caf31f552c9c4879b15865d38ba1dfb0f7a7d2ac46a4f0cae25ba"),
+  },
+};
+
+const BLOCKLOCK_IBE_OPTS: IbeOpts = {
+  hash: keccak_256,
+  k: 128,
+  expand_fn: "xmd",
+  dsts: {
+    H1_G1: Buffer.from("BLOCKLOCK_BN254G1_XMD:KECCAK-256_SVDW_RO_H1_"),
+    H2: Buffer.from("BLOCKLOCK_BN254_XMD:KECCAK-256_H2_"),
+    H3: Buffer.from("BLOCKLOCK_BN254_XMD:KECCAK-256_H3_"),
+    H4: Buffer.from("BLOCKLOCK_BN254_XMD:KECCAK-256_H4_"),
+  },
+};
+
+const blsKey = "0x58aabbe98959c4dcb96c44c53be7e3bb980791fc7a9e03445c4af612a45ac906";
+const SCHEME_ID = "BN254-BLS-BLOCKLOCK";
+
+let server: any;
+let provider: ethers.JsonRpcProvider;
+let wallet: ethers.Signer;
+let accounts: ethers.Signer[];
+
+describe("Blocklock blockchain integration tests with Ganache", () => {
+  beforeAll(async () => {
+    // Start Ganache server with a mnemonic
+    server = Ganache.server({
```

Comment:
not important, but any reason for ganache over anvil? I thought ganache was EOL

Reply:
I did a quick check and didn't find a way to setup Anvil programmatically. But I think there might be a solution around it. I will check and benchmark against ganache in a separate test file and keep the faster one in the next release.